### DISCUSSION TITLE
Maximum number of packages in an SBOM

### DISCUSSION THREAD

**Discussion by mrmoosefish** [DependencyTrack/dependency-track#1679] [Q&A]
I attempted to import an SBO with about 6,000 packages in it. The import fails immediately with a pretty vague message.

Is there an upper limit to the number of packages that can be imported in a single SBOM or is it dependent on available memory? If it's limited, can an SBOM be broken into multiple pieces and loaded into the same project in chunks? When I tried this, I found it essentially replaced the components rather than adding them.

**Comment by nscuro** [comment]
Can you share some of those vague messages? That would help with understanding what's going on.

Theoretically yes, the maximum size of BOMs being ingestable somewhat depends on your RAM. However, it's unlikely that you hit this hypothetical limit with common RAM configurations these days.

**Reply by mrmoosefish** [reply]
The error message that pops in the upper right corner reads, "Could not perform the requested action." (Did I mention it's vague?)

One challenge in troubleshooting this is I don't have access to the container and don't see a means to forward logs to our SIEM. Is that configurable
![Screen Shot 2022-06-01 at 2 03 42 PM](https://user-images.githubusercontent.com/75909952/171501615-94b85f8f-10b0-41fb-99c7-63730b6be98c.png)
?

**Reply by stevespringett** [reply]
What does the dependency-track log say when the bom is uploaded? The biggest BOM that I've been able to create was just over 15K components and did not have an issue. But the log should give us a clue to what's happening.

**Reply by itsmemail4u** [reply]
actually i am also facing same problem. i am using DT 4.4.2  version and whenever try to import file having >=3000 packages, error appears. even i can not append the packages in exiting projects. it always overwrite. 
<img width="957" alt="DT_Error" src="https://user-images.githubusercontent.com/61532520/172376731-455d044f-f5f9-4905-9167-478870254016.png">

**Reply by mrmoosefish** [reply]
I've been able to review three logs.

docker_api.log shows me creating a new project, but then no references to uploading the SBOM.

> 2022-06-14 23:13:13,255 INFO [ProjectResource] Project chik 3 : 3 created by jjsoltys

dtrack_frontend.log shows some activity immediately before the project creation, but nothing after.

> x.x.x.x - - [14/Jun/2022:23:12:46 +0000] "GET /js/chunk-2d64c586.79ab7ecb.js HTTP/1.1" 200 204372 "https://xx.xx.xx/projects" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.0.0 Safari/537.36" "x.x.x.x"

dtrack_nginx_proxy.log shows nothing related to my activity.

Is there another log I should be looking in?
